### PR TITLE
Fix AudioClip I/O

### PR DESCRIPTION
**Pull request by MaigoAkisame**
The existing code does not read and write audio signals without changing them.
I've fixed some of the related code, and added a `test_audioclip_io` test to ensure the audio does not change during I/O.

- [X] I have provided code that clearly demonstrates the bug and only works correctly when used with this PR
- [X] I have added suitable tests to the test suite in `tests/`
- [x] I have properly documented new or changed features in the documention or in the docstrings
- [x] I have properly documented unusual changes to the code in the comments around it
- [x] I have made note of any breaking/backwards incompatible changes

### CODE REVIEW

File: moviepy/audio/io/readers.py [hunk 1]
```diff
@@ -119,6 +119,9 @@ def read_chunk(self,chunksize):
         result = (1.0*result / 2**(8*self.nbytes-1)).\
                                  reshape((int(len(result)/self.nchannels),
                                           self.nchannels))
+        pad = np.zeros((chunksize - len(result), self.nchannels),
+                       dtype=result.dtype)
+        result = np.concatenate([result, pad])
```
Comment (MaigoAkisame) [inline moviepy/audio/io/readers.py]: Pad the read chunk with zeros when there isn't enough audio left to read, so the buffer is always at full length.

**Comment by MaigoAkisame** [comment]
All checks have passed!

What's the next step to have this merged?

(It's my first time to make a pull request, so forgive me if I'm asking anything stupid)

**Comment by tburrows13** [comment]
Hey @MaigoAkisame thanks so much for making this PR! You've done everything you need to for now, it is just waiting for a maintainer to check that we're happy with it and then we can merge it into the master branch.

**Comment by MaigoAkisame** [comment]
Thank you!

File: tests/test_AudioClips.py [hunk 2]
```diff
@@ -35,6 +37,23 @@ def test_audioclip():
...
+    clip = AudioArrayClip(input_array, fps=44100)
+    clip.write_audiofile(os.path.join(TMP_DIR, "random.wav"))
+    # Load the clip.
+    # The loaded clip will be slightly longer because the duration is rounded
+    # up to 4.99 seconds.
+    # Verify that the extra frames are all zero, and the remainder is identical
+    # to the original signal.
+    clip = AudioFileClip(os.path.join(TMP_DIR, "random.wav"))
+    output_array = clip.to_soundarray()
+    assert np.abs(output_array[: len(input_array)] - input_array).max() < 4e-5
```
Comment (mgaitan) [inline tests/test_AudioClips.py]: what about to use numpy's [assert_array_almost_equal] (https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_array_almost_equal.html#numpy.testing.assert_array_almost_equal). 

It's the canonical way to compare arrays.  

```suggestion
    assert_array_almost_equal(output_array[:len(input_array)], input_array, decimal=4) 
```

**Comment by MaigoAkisame** [comment]
There has been no activity for almost 2 weeks. Is someone gonna merge this?

File: moviepy/audio/io/readers.py [hunk 3]
```diff
@@ -239,7 +239,7 @@ def buffer_around(self, framenumber):
             current_f_end = self.buffer_startframe + self.buffersize
             if new_bufferstart < current_f_end < new_bufferstart + self.buffersize:
                 # We already have one bit of what must be read
-                conserved = current_f_end - new_bufferstart + 1
+                conserved = current_f_end - new_bufferstart
```
Comment (tburrows13) [inline moviepy/audio/io/readers.py:242]: The comment on line 241 seems to be talking about the `+ 1` that you've removed. Presumably the comment should be deleted as well if you've decided that it's unnecessary?

Comment (MaigoAkisame) [reply moviepy/audio/io/readers.py:242]: "One bit" sounds ambiguous. I changed that to "part" to clarify.

Comment (tburrows13) [reply moviepy/audio/io/readers.py:242]: Yes, that makes more sense, thanks!

**Comment by tburrows13** [comment]
Closes #1174. Thanks

**Comment by tburrows13** [comment]
Sorry, my mistake. Please merge the changes back in at https://github.com/MaigoAkisame/moviepy/pull/1.

**Comment by MaigoAkisame** [comment]
Is that something I need to do? What do I need to do exactly?

**Comment by tburrows13** [comment]
Please just merge https://github.com/MaigoAkisame/moviepy/pull/1, and I think that I can sort the rest out after that.

**Comment by MaigoAkisame** [comment]
Sure! I just did it.

**Comment by tburrows13** [comment]
Great, I'll merge when the tests finish. You'll be happy to know that this PR fixed #1174 perfectly.

**Pull request merged**